jsk_fetch_robot/fetcheus/fetch-interface.l

```diff
@@ -40,8 +40,11 @@
      (ros::warn ":angle-vector tm is not a number, use :angle-vector av tm args"))
    (let ((use-torso t))
      (if (and (member :use-torso args) (null (cadr (member :use-torso args)))) (setq use-torso nil))
-     (send self :angle-vector-motion-plan av :move-arm :rarm :total-time tm :start-offset-time 0.01 :use-torso use-torso :clear-velocities t)
+     (send* self :angle-vector-motion-plan av :move-arm :rarm :total-time tm :start-offset-time 0.01 :use-torso use-torso :clear-velocities t args)
```

Comment:
I understood the reasons why it should be better not to include `:rarm-angle-vector` and agree with it. About the controller correction, I was thinking about starting a new pull request for it, since there are actually a lot of problems with the code above and the discussion got a bit messy.

However, when I tried defining as [baxter-interface.l](https://github.com/jsk-ros-pkg/jsk_robot/blob/master/jsk_baxter_robot/baxtereus/baxter-interface.l#L237), i.e. setting ctype as optional parameter and sending it as a key to :angle-vector-motion-plan, I faced the following problems:

■ Specifying controllers doesn't give expected result
- The arm always move
- Torso moves when :use-torso is set as t (default), and doesn't when set as nil

Detailed behaviour can be verified at [this](https://drive.google.com/open?id=0B6dW2G6plvj7eF9zM29UQzBJSXM) capture.

Above result was:
```
113.E8-irteusgl$ symbol-value 'coords
#f(100.0 80.9765 86.9744 -71.7262 121.91 -136.127 41.4067 -39.0282 -0.000253 44.2877)
114.E8-irteusgl$ test send *ri* :angle-vector coords 3000 :args :default-controller :arm-controller (:arm-controller :use-torso nil) :head-controller (:head-controller :use-torso nil) :torso-controller
Evaluating: (send *ri* :angle-vector coords 3000 :default-controller) ...

[ INFO] [1494313876.289271251, 2232.369000000]: ;; Planned Trajectory Total Time   2.607 [sec]
[ INFO] [1494313876.291028458, 2232.370000000]: ;; Scaled Trajectory Total Time   3.000 [sec]
[ INFO] [1494313876.291088349, 2232.370000000]: ;; generated 17 points for 3 sec using [arm_with_torso] group
	=> (#f(100.0 80.9765 86.9744 -71.7262 121.91 -136.127 41.4067 -39.0282 -0.000253 44.2877))
[ INFO] [1494313880.422863395, 2235.633000000]: ;; Planned Trajectory Total Time   2.607 [sec]
[ INFO] [1494313880.423279213, 2235.633000000]: ;; Scaled Trajectory Total Time   3.000 [sec]
[ INFO] [1494313880.423348735, 2235.633000000]: ;; generated 17 points for 3 sec using [arm_with_torso] group
Evaluating: (send *ri* :angle-vector coords 3000 :arm-controller) ...

[ INFO] [1494313885.926336314, 2239.762000000]: ;; Planned Trajectory Total Time   2.606 [sec]
[ INFO] [1494313885.926818746, 2239.762000000]: ;; Scaled Trajectory Total Time   3.000 [sec]
[ INFO] [1494313885.926848678, 2239.762000000]: ;; generated 17 points for 3 sec using [arm_with_torso] group
	=> (#f(100.0 80.9765 86.9744 -71.7262 121.91 -136.127 41.4067 -39.0282 -0.000253 44.2877))
[ INFO] [1494313889.955850864, 2243.000000000]: ;; Planned Trajectory Total Time   2.729 [sec]
[ INFO] [1494313889.956356734, 2243.000000000]: ;; Scaled Trajectory Total Time   3.000 [sec]
[ INFO] [1494313889.956404325, 2243.000000000]: ;; generated 18 points for 3 sec using [arm_with_torso] group
Evaluating: (send *ri* :angle-vector coords 3000 :arm-controller :use-torso nil) ...

[ INFO] [1494313895.459078421, 2247.216000000]: ;; Planned Trajectory Total Time   2.619 [sec]
[ INFO] [1494313895.459459834, 2247.216000000]: ;; Scaled Trajectory Total Time   3.000 [sec]
[ INFO] [1494313895.459493903, 2247.216000000]: ;; generated 17 points for 3 sec using [arm] group
	=> (#f(100.0 80.9765 86.9744 -71.7262 121.91 -136.127 41.4067 -39.0282 -0.000253 44.2877))
[ INFO] [1494313899.516929054, 2250.489000000]: ;; Planned Trajectory Total Time   2.752 [sec]
[ INFO] [1494313899.517869587, 2250.490000000]: ;; Scaled Trajectory Total Time   3.000 [sec]
[ INFO] [1494313899.517928325, 2250.490000000]: ;; generated 18 points for 3 sec using [arm_with_torso] group
;; #<linear-joint #X603b198 l_gripper_finger_joint> :joint-angle(50.0942) violate max-angle(50.0)
Evaluating: (send *ri* :angle-vector coords 3000 :head-controller) ...

[ INFO] [1494313904.939922436, 2254.665000000]: ;; Planned Trajectory Total Time   2.606 [sec]
[ INFO] [1494313904.940859718, 2254.666000000]: ;; Scaled Trajectory Total Time   3.000 [sec]
[ INFO] [1494313904.940904955, 2254.666000000]: ;; generated 17 points for 3 sec using [arm_with_torso] group
	=> (#f(100.0 80.9765 86.9744 -71.7262 121.91 -136.127 41.4067 -39.0282 -0.000253 44.2877))
[ INFO] [1494313909.048225374, 2257.947000000]: ;; Planned Trajectory Total Time   2.606 [sec]
[ INFO] [1494313909.048844030, 2257.947000000]: ;; Scaled Trajectory Total Time   3.000 [sec]
[ INFO] [1494313909.048949346, 2257.947000000]: ;; generated 17 points for 3 sec using [arm_with_torso] group
;; #<linear-joint #X603b198 l_gripper_finger_joint> :joint-angle(50.1017) violate max-angle(50.0)
Evaluating: (send *ri* :angle-vector coords 3000 :head-controller :use-torso nil) ...

[ INFO] [1494313914.669466487, 2262.185000000]: ;; Planned Trajectory Total Time   2.622 [sec]
[ INFO] [1494313914.669986703, 2262.185000000]: ;; Scaled Trajectory Total Time   3.000 [sec]
[ INFO] [1494313914.670526769, 2262.185000000]: ;; generated 17 points for 3 sec using [arm] group
	=> (#f(100.0 80.9765 86.9744 -71.7262 121.91 -136.127 41.4067 -39.0282 -0.000253 44.2877))
[ INFO] [1494313918.779124456, 2265.419000000]: ;; Planned Trajectory Total Time   2.629 [sec]
[ INFO] [1494313918.779503030, 2265.419000000]: ;; Scaled Trajectory Total Time   3.000 [sec]
[ INFO] [1494313918.779556053, 2265.419000000]: ;; generated 17 points for 3 sec using [arm_with_torso] group
Evaluating: (send *ri* :angle-vector coords 3000 :torso-controller) ...

[ INFO] [1494313924.357370970, 2269.645000000]: ;; Planned Trajectory Total Time   2.607 [sec]
[ INFO] [1494313924.357721528, 2269.645000000]: ;; Scaled Trajectory Total Time   3.000 [sec]
[ INFO] [1494313924.357743846, 2269.645000000]: ;; generated 17 points for 3 sec using [arm_with_torso] group
	=> (#f(100.0 80.9765 86.9744 -71.7262 121.91 -136.127 41.4067 -39.0282 -0.000253 44.2877))
[ INFO] [1494313928.391542859, 2272.897000000]: ;; Planned Trajectory Total Time   2.607 [sec]
[ INFO] [1494313928.393715246, 2272.898000000]: ;; Scaled Trajectory Total Time   3.000 [sec]
[ INFO] [1494313928.393856807, 2272.898000000]: ;; generated 17 points for 3 sec using [arm_with_torso] group
;; #<linear-joint #X603b198 l_gripper_finger_joint> :joint-angle(50.0885) violate max-angle(50.0)
t
115.E8-irteusgl$
```

■ Head movement doesn't seem to be included on collision check

#775


■ Cannot assign :use-torso without setting controller

Could be a problem since it would require changing existing codes.


If my understanding is correct could set :use-torso to nil for :arm-controller and use :angle-vector-raw for :head-controller and :torso-controller, but this way collisions might occur.